### ISSUE TITLE
Inconsistent dimension name for string length between gas optics NetCDF files

### ISSUE THREAD

**Issue by Chiil** [earth-system-radiation/rte-rrtmgp#103]
The dimension indicating string length is named `string_len` in the longwave gas optics coefficients file, but `string32` in the shortwave file. This results in some complications in a generic reader. In earlier versions both were named `string_len`. It would be great if that could be restored.

**Comment by RobertPincus** [comment]
@pernak18 I'm hoping you can use some Python-fu here? Roughly we want a PR with a new SW coefficients file that has precisely the same structure but a revised dimension name consistent with the LW file.

**Comment by pernak18** [comment]
@RobertPincus sounds doable, but will have to wait until i have some free time. i'm a pretty small line item in Eli's budget for April

**Comment by pernak18** [comment]
problem was more complex than i thought it would be, but the code i used at NERSC (`cori`) is:

```
#!/usr/bin/env python

import os, sys

PIPPATH = '{}/.local/'.format(os.path.expanduser('~')) + \
    'cori/3.7-anaconda-2019.10/lib/python3.7/site-packages'
paths = [PIPPATH, '../common']
for path in paths: sys.path.append(path)

import xarray as xa

strVars = ['gas_minor', 'gas_names', 'identifier_minor', 
           'minor_gases_lower', 'minor_gases_upper', 
           'scaling_gas_lower', 'scaling_gas_upper']

ncFile = 'rrtmgp-data-sw-g224-2018-12-04.nc'

with xa.open_dataset(ncFile, concat_characters=False) as ds:
    outDS = ds.rename({'string32': 'string_len'})

# https://github.com/pydata/xarray/issues/2899
# workaround actually worked for me
outNC = r'string_len_{}'.format(ncFile)
encode = {}
for sv in strVars: encode[sv] = {'dtype': 'str'}
outDS.to_netcdf(outNC, encoding=encode)

print('Wrote {}'.format(outNC))
```

i apparently cannot attach the resulting file in this issue. do we want this code somewhere in version control, or does the code snippet suffice?

**Comment by RobertPincus** [comment]
This seems awesome, @pernak18, thanks so much. Can you open a PR against `develop` in which you update `rrtmgp/data/rrtmgp-data-sw-g224-2018-12-04.nc` to use the file with the renamed dimension?  Having the code in the issue and/or PR is enough, I think.

**Comment by pernak18** [comment]
for my reference:

```
% pwd
/global/homes/p/pernak18/RRTMGP/rte-rrtmgp
% git checkout develop
Branch 'develop' set up to track remote branch 'develop' from 'origin'.
Switched to a new branch 'develop'
% cd rrtmgp/data/
% cp ~/RRTMGP/g-point-reduction/dim_swap/modified_rrtmgp-data-sw-g224-2018-12-04.nc rrtmgp-data-sw-g224-2018-12-04.nc 
cp: overwrite 'rrtmgp-data-sw-g224-2018-12-04.nc'? y
```

on `cori`

**Comment by Chiil** [comment]
The files in the new `main` repository have inconsistent `NetCDF` types:
```
chiel@MacBookChiel:~/devel/radiation/rte-rrtmgp/rrtmgp/data$ ncdump -k rrtmgp-data-sw-g224-2018-12-04.nc 
netCDF-4
chiel@MacBookChiel:~/devel/radiation/rte-rrtmgp/rrtmgp/data$ ncdump -k rrtmgp-data-lw-g256-2018-12-04.nc 
classic
```

**Comment by RobertPincus** [comment]
<facepalm> thanks for noticing

**Comment by pernak18** [comment]
@RobertPincus does this mean we want to switch LW to `netCDF-4` or SW to `classic`?

**Comment by RobertPincus** [comment]
@pernak18 I'm not sure it matters, `netCDF-4` is the future though. @Chiil do you have an opinion?

**Comment by m214089** [comment]
HI,

I do not think that netcdf-4 is the future after the hdf group changed there working model. I think for such small files the netcdf classic format (aka netcdf-3 is best.
Cheerio,
Luis

**Comment by pernak18** [comment]
i've converted SW to `classic`. `xarray` has 4 different options, and i initially went with `NETCDF4_CLASSIC` as a combination of both but then decided to stay consistent with the LW, so:

`outDS.to_netcdf(outNC, encoding=encode, format='NETCDF3_CLASSIC')`

**Comment by RobertPincus** [comment]
Fixed in commit `cd7a6c2` (#109)
